server/hl7/hl7.controller.js

```diff
@@ -24,27 +29,61 @@ const upload = multer({ storage,
 
 
 /**
- * Creates a new Mongoose object for the rile to save in the DB
- * We do this mainly to avoid the (new-cap) rule error from ES6
+ * Takes in a raw hl7 message or a list of raw messages and returns a list of the parsed message
  */
-function newHl7File(fileName) {
-  return new HL7({ fileName });
+function parseRawHl7(rawHl7Message) {
+  // TODO: Implement this function
+  return {
+    ADT: rawHl7Message
+  };
+}
+
+/**
+ * Reads a file from a specified path and returns a list of hl7 messages
+ * @param filePath
+ * @returns {*}
+ */
+function readFile(filePath) {
+  // TODO: Aditya move your code in here and modify accordingly.
+  return filePath;
 }
 
 /**
  * Utilizing Multer, this function receives a file as a request, and saves the generated file name
  * to the DB.
  */
-function uploadFile(req, res, next) {
-  const hl7File = newHl7File(req.file.filename);
-
-  hl7File.save()
-    .then(() => res.status(httpStatus.CREATED).json(hl7File))
-    .catch(() => {
-      const err = new APIError(`Error: Failed to save file ${req.file.filename}`, httpStatus.BAD_REQUEST);
-      next(err);
-    });
-}
+function parseFile(req, res, next) {
+  new Promise((resolve, reject) => {
+    fs.readFile(req.file.path, 'utf8', (fsErr, data) =>
+      (fsErr ? reject(fsErr) : resolve(data))
+    );
+  }).catch(err =>
+    // At this point, even if the file isn't read here, it has
+    // already been saved to the file system. So we would want to
+    // handle that situation somehow.
+    // TODO: Handle case where file isn't read but has been saved to the FS.
+    next(new APIError(err, httpStatus.BAD_REQUEST))
+  ).then((data) => {
+    req.user.files.unshift({ filename: req.file.path });
+    const newFile = req.user.files[0];
+    return Promise.all([data, newFile, req.user.save()]);
+  }).then(([data, file]) => {
+    const hl7MessageList = data
+      .replace(/\n\r/g, '\n')
+      .replace(/\r/g, '\n')
+      .split(/\n{2,}/g);
 
+    return Message.create(hl7MessageList.filter(rawMessage => !!rawMessage)
```

Comment:
It doesn't seem like we save meta data about the uploaded file like we were doing before. Is that something we still want to do?

Reply:
it is now within the user's document!